File: hello-libs/app/src/main/java/com/example/hellolibs/MainActivity.java
package com.example.hellolibs;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.TextView;
/*
 * Simple Java UI to trigger jni function. It is exactly same as Java code
 * in hello-jni.
 */
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        TextView tv = new TextView(this);
        tv.setText( "The calculation took " + measureTicks() + " ticks" );
        setContentView(tv);
    }
    public native long  measureTicks();
    static {
        System.loadLibrary("hello-libs");
    }

}

File: prefab/prefab-dependency/app/src/main/java/com/example/prefabdependency/MainActivity.kt
package com.example.prefabdependency

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.example.prefabdependency.databinding.ActivityJsonParseBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityJsonParseBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityJsonParseBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)
        binding.resultTextview.text = getJsonValue(
            """
            {
                "cpp": "clang",
                "aar": "Prefab",
                "package": "cmake",
                "app": "gradle"
            }
            """.trimIndent(), "aar"
        )

    }

    private external fun getJsonValue(jsonString: String, key: String): String

    companion object {
        init {
            System.loadLibrary("app")
        }
    }
}
